src/urllib3/exceptions.py

```diff
@@ -32,9 +32,9 @@ class HTTPWarning(Warning):
 class PoolError(HTTPError):
     """Base exception for errors caused within a pool."""
 
-    pool: "ConnectionPool"
+    pool: Optional["ConnectionPool"]
 
-    def __init__(self, pool: "ConnectionPool", message: str) -> None:
+    def __init__(self, pool: Optional["ConnectionPool"], message: str) -> None:
```

Comment:
Given that we only need to change this due to the tests, I infer that the actual code never passes `None` here, so it's better to fix the tests instead of pessimizing the type for all users.

Maybe adding 

```py
DUMMY_POOL = ConnectionPool("dummy")
```

to the test file and passing it instead of `None`?

Reply:
Great suggestion. Thanks!

Reply:
This was the suggestion I wanted implemented :)